### PR TITLE
Mention where `std::vec` structs originate from.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Comment by TimNN** [comment]
This patch causes a link check failure, see #34819.

Excerpt from the log:

```
Linkcheck stage2 (x86_64-pc-windows-msvc)
collections\vec\struct.IntoIter.html:55: broken link - collections\vec\provided by
the [`IntoIterator`] trait
std\vec\struct.IntoIter.html:55: broken link - std\vec\provided by
the [`IntoIterator`] trait


command did not execute successfully: "C:\\bot\\slave\\auto-win-msvc-64-opt-rustbuild\\build\\obj\\build\\x86_64-pc-windows-msvc\\stage2-tools\\x86_64-pc-windows-msvc\\release\\linkchecker.exe" "C:\\bot\\slave\\auto-win-msvc-64-opt-rustbuild\\build\\obj\\build\\x86_64-pc-windows-msvc\\doc"
expected success, got: exit code: 101
```

File: src/libcollections/vec.rs [hunk 1]
```diff
@@ -1603,6 +1603,12 @@ impl<'a, T> FromIterator<T> for Cow<'a, [T]> where T: Clone {
 ////////////////////////////////////////////////////////////////////////////////
 
 /// An iterator that moves out of a vector.
+///
+/// This `struct` is created by the `into_iter` method on [`Vec`][`Vec`] (provided
+/// by the [`IntoIterator`] trait).
+///
+/// [`Vec`]: struct.Vec.html
+/// [`IntoIterator`]: ../iter/trait.IntoIterator.html
```
Comment (ollie27) [inline src/libcollections/vec.rs]: You'll need to change this to `../../std/iter/trait.IntoIterator.html` so the link works from the `collections` docs as well.

Comment (frewsxcv) [inline src/libcollections/vec.rs]: Good call, didn't think about the collection docs getting generated. Will force push shortly.

Comment (frewsxcv) [inline src/libcollections/vec.rs]: Latest force push includes this change.

**Pull request merged**